component-framework/TS_PropertySetTableControl/TSPropertySetTableControl/index.ts

```diff
@@ -0,0 +1,355 @@
+		if (!context.parameters.sampleDataSet.columns) {
+			return [];
+		}
+
+		let columns = context.parameters.sampleDataSet.columns;
+
+		return columns;
+	}
+
+	/**
+	 * Get column width distribution
+	 * @param context context object of this cycle
+	 * @param columnsOnView columns array on the configured view
+	 * @returns column width distribution
+	 */
+	private getColumnWidthDistribution(context: ComponentFramework.Context<IInputs>, columnsOnView: DataSetInterfaces.Column[]): string[] {
+
+		let widthDistribution: string[] = [];
+
+		// Considering need to remove border & padding length
+		let totalWidth: number = context.mode.allocatedWidth;
+		let widthSum = 0;
+
+		columnsOnView.forEach(function (columnItem) {
+			widthSum += columnItem.visualSizeFactor;
+		});
+
+		let remainWidth: number = totalWidth;
+
+		columnsOnView.forEach(function (item, index) {
+			let widthPerCell = "";
+			if (index !== columnsOnView.length - 1) {
+				let cellWidth = Math.round((item.visualSizeFactor / widthSum) * totalWidth);
+				remainWidth = remainWidth - cellWidth;
+				widthPerCell = cellWidth + "px";
+			}
+			else {
+				widthPerCell = remainWidth + "px";
+			}
+			widthDistribution.push(widthPerCell);
+		});
+
+		return widthDistribution;
+
+	}
+
+	private createTableHeader(columnsOnView: DataSetInterfaces.Column[], widthDistribution: string[]): HTMLTableSectionElement {
+
+		let tableHeader: HTMLTableSectionElement = document.createElement("thead");
+		let tableHeaderRow: HTMLTableRowElement = document.createElement("tr");
+		tableHeaderRow.classList.add("SimpleTable_TableRow_Style");
+		const _this = this;
+		columnsOnView.forEach(function (columnItem, index) {
+			let tableHeaderCell = document.createElement("th");
+			tableHeaderCell.classList.add("SimpleTable_TableHeader_Style");
+			let innerDiv = document.createElement("div");
+			innerDiv.classList.add("SimpleTable_TableCellInnerDiv_Style");
+			innerDiv.style.maxWidth = widthDistribution[index];
+			let columnDisplayName: string;
+			if (columnItem.order < 0) {
+				columnDisplayName = columnItem.displayName + "(propertySet)";
+			} else {
+				columnDisplayName = columnItem.displayName;
+			}
+			innerDiv.innerText = columnDisplayName;
+
+			tableHeaderCell.appendChild(innerDiv);
+			tableHeaderRow.appendChild(tableHeaderCell);
+		});
+
+		tableHeader.appendChild(tableHeaderRow);
+		return tableHeader;
+	}
+
+	private createSearchBar(context: ComponentFramework.Context<IInputs>) {
+		let container = document.createElement("div");
+		let input = document.createElement("input");
+		input.id = "searchBar";
+		let button = document.createElement("button");
+		button.innerHTML = "Search"
+		button.addEventListener("click", (() => {
+			let conditionsArray: DataSetInterfaces.ConditionExpression[] = [];
+			let searchString = input.value;
+			for (let i = 0; i < this.columns.length; i++) {
+				const column = this.columns[i];
+				if (!column.isHidden && column.dataType === "SingleLine.Text") {
```

Comment:
Why only string ? so this search is different from the quick find and using advanced find ?

Reply:
This part will be removed due to filter is not in prod yet.
The reason we limit to string in this sample is due to the filter query limitation in Canvas. The operator we used in this sample, LIKE, will be mapped to beginsWith, which works only on strings. Trying to use this operator on any other data types would result query error.